### PR TITLE
[dynamo] Skip patch function for lr scheduler tracing

### PR DESCRIPTION
**Pull request by mlazos**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #170019
* #169989
* #169988
* #169987

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo

### CODE REVIEW

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -391,6 +391,13 @@ def call_function(
         args: Sequence[VariableTracker],
         kwargs: dict[str, VariableTracker],
     ) -> VariableTracker:
+        # Ignore patch_track_step_called from torch/optim/lr_scheduler.py - it just patches
+        # the optimizer.step method and we don't need to trace it
+        if (
+            self.get_name() == "patch_track_step_called"
+            and self.get_filename().endswith("torch/optim/lr_scheduler.py")
+        ):
```
Comment (williamwen42) [inline torch/_dynamo/variables/functions.py:399]: Could we refer to this function directly rather than check for name?

Comment (mlazos) [reply torch/_dynamo/variables/functions.py:399]: it's a local function unfortunately which makes it even more annoying. Hence why I check file and name (tried to narrow the scope as much as possible)

**Review by williamwen42 — APPROVED** [review]
Kinda hacky, but I believe it should be possible to extract the code object for the local function that we could compare to - up to you if you want to try this though.
